src/docs/development/tools/sdk/upgrading.md

```diff
@@ -29,36 +26,85 @@ dev_dependencies:
     sdk: flutter
 ```
 
-The `sdk: flutter` line tells the `flutter` command-line tool to find the
-correct package for you.
-
-Do not use the `pub get` or `pub upgrade` commands to manage your dependencies.
+{{site.alert.warning}}
+Don't use the `pub get` or `pub upgrade` commands to manage dependencies
+for Flutter apps.
 Instead, use `flutter packages get` or `flutter packages upgrade`.
 If you want to use pub manually, you can run it directly by setting the
 `FLUTTER_ROOT` environment variable.
+{{site.alert.end}}
+
 
-## Upgrading Flutter channel and your packages
+## Upgrading the Flutter SDK and packages
 
-To update both the Flutter SDK and your packages, use the `flutter upgrade`
-command from the root of your app (the same directory that contains the
-`pubspec.yaml` file):
+To update both the Flutter SDK and the packages that your app depends on,
+use the `flutter upgrade` command from the root of your app
+(the same directory that contains the `pubspec.yaml` file):
 
 ```terminal
 $ flutter upgrade
 ```
 
-## Upgrading your packages
+This command first gets the most recent version of the Flutter SDK
+that's available on your Flutter channel.
+Then this command updates each package that your app depends on
+to the most recent compatible version.
+
+If you want an even more recent version of the Flutter SDK,
+switch to a less stable Flutter channel
+and then run `flutter upgrade`.
+
+### Switching your Flutter channel
```

Comment:
- Consider making this a level 2 heading.
- Switching your Flutter channel -> Switching Flutter channels (?)